### ISSUE TITLE
Run on specified networks does not work 

### ISSUE THREAD

**Issue by ushai9726** [syncthing/syncthing-android#1775]
When I try to set specific networks Syncthing will work on, I just get a pop-up saying connect to a network to add it to the list. 

I am connected to a Wifi network, but it is not added to a list. The setting is not changed, nor any list of networks appear. 

If I disconnect from the wifi network and reconnect, nothing happens 

### Version Information

    App Version: 1.19.0
    Syncthing Version: v1.19.0.3
    Android Version: Android 9

**Comment by ushai9726** [comment]
![Screenshot_20220330-114314_Syncthing](https://user-images.githubusercontent.com/102657467/160794666-5c6cbb0b-75d4-4a31-9a85-0ca6461f0d29.jpg)

**Comment by Knud3** [comment]
Also I have this problem with Samsung S22. Works with S10 with same version.

I have manually selected location permission as "Allow all the time".

Also when trying to disable precise location Syncthing keeps asking location permission even it is given. Does it need precise location?

App Version: 1.19.2
Syncthing Version: v1.19.2
Android Version: Android 12 (newest firmware)

**Comment by imsodin** [comment]
@Knud3 This issue isn't about location permissions and your requests sounds like a question for support on how to configure permissions with Syncthing in general. Please ask such questions on https://forum.syncthing.net instead of the issue tracker. There's more friendly eyes to chime in and help you there. Thanks!

**Comment by Knud3** [comment]
@imsodin What do you mean? I can't select a specific Wi-Fi network no matter what permission is given. That is an issue, right?

An additional (more generic) question was, that why it asks location permission even when location permission is given.

My English is not the best, so let me illustrate the problem very simply:
- Location permission: Don't allow > Syncthing gives "You need to grant LOCATION permission to use this feature."
- Location permission: Allow all the time, Use precise location: Disabled > Syncthing gives "You need to grant LOCATION permission to use this feature."
- **Location permission: Allow all the time, Use precise location: Enabled > Syncthing gives "Please connect to a Wi-Fi to add it to the list"**

And yes, it is connected to Wi-Fi all the time.

**Comment by imsodin** [comment]
Thanks for the clarification: So you are experiencing the same bug. My reaction was about the more generic questions: Please direct those to https://forum.syncthing.net.

**Comment by felsgaertner** [comment]
@Knud3 Thanks for pointing that out - I had the same problem, and after enabling android's location, I could select networks.

So that seems to be not an error, but kind of missing documentation: could you mention in the app that it needs location permission **and** location has to be turned on?

**Comment by Knud3** [comment]
I actually got this working without location permission or location enabled.

With help of Tasker I set up profile which send `com.nutomic.syncthingandroid.action.START` intend when wpa_supplicant component give `'CTRL-EVENT-CONNECTED'.*'WIFINAME` logcat entry.

Of course it is little more advanced setup, but for me thats more than ok, because I already use Tasker for ton of stuff.

Tasker also need some ADB commands to get permission to read log entries. https://tasker.joaoapps.com/userguide/en/help/ah_read_logs_grant.html

**Comment by ecanja** [comment]
Any plans to fix this? Unfortunately, that's still an unsolved bug.

@felsgaertner I wouldn't classify this as a documentation issue. Sure, the fact that you have to turn on location for this to work (why actually?) is missing in the documentation and should be there. But the problem @Knud3 pointed out is still there:

> I found out that if I turn location on, then it starts working.
> 
> Also after Wi-Fi is selected and I disable location and/or any of location permission it stops working and do not work even when everything is set back to where it was. I need to de-select Wi-Fi network and select it again.

I also configured a specific Wi-Fi (with location on) and it worked, but after I turn off the Wi-Fi and turn it back on, it won't work anymore. I get the error message

```
Syncthing is disabled
[...]
Reasons:
Current WiFi SSID is not whitelisted
```

I can go to the settings and whitelist the exact same Wi-Fi that's already there in the settings and then it works again. But of course I don't want to do this every time when I turn (back) on the Wi-Fi. But if I don't, synchronization will stop. That's why I would declare that a bug.

**Comment by Knud3** [comment]
I had to change to Catfriend1 fork of Syncthing Android, because I also had problem with VPN that everytime I leave/enter my home it won't find my network without going to settings and do nothing (yes, just open settings and it started to work).

**Comment by LinuxOnTheDesktop** [comment]
Let me add - though perhaps I should create a separate issue for this - that the UI seems to add insult to injury. The injury is the failure to detect a whitelisted network. The insult (as it were) is that it is hard to determine how to work around the problem. For, it seems that the procedure to disable whitelisting is counter-intuitive. That procedure, I take it, is as follows (at least if one wants to retain some other - non-wifi - run-condition). One unticks each of any network boxes of this sort:

![image](https://user-images.githubusercontent.com/47056543/207461827-a0836bea-8e25-470b-abdd-3c701bea406c.png)

**Comment by LinuxOnTheDesktop** [comment]
It would be good to have someone 'assigned' to this, or a 'milestone' assigned.

**Comment by tomasz1986** [comment]
I don't believe issues are "assigned" here, i.e. anyone can work on a particular one as long as they're willing to do so. As for the milestone, if no-one is working on the issue, there can be no milestone provided, as nobody can guarantee when the issue gets actually fixed.

**Comment by imsodin** [comment]
Mainly: https://github.com/syncthing/syncthing/wiki/NoPlusOne
Also: https://github.com/syncthing/syncthing-android?tab=readme-ov-file#status-maintenance-mode---co-maintainers-welcome

**Comment by quite** [comment]
@ecanja 

> Sure, the fact that you have to turn on location for this to work (why actually?) is missing in the documentation and should be added. 

An app needs to be given the Location permission in order to get hold of details about the Wifi/WLAN it is conncted to, because the physical (relative) location could be inferred from those.
